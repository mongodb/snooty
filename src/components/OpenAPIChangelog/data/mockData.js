export const mockChangelog = [
  {
    date: '2024-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'removed',
            changes: [
              {
                change: 'resource version 2023-01-01 was removed, latest available resource version is 2023-02-01',
                changeCode: 'resource-version-removed',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-01',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/alertConfigs/{alertConfigId}',
        httpMethod: 'PUT',
        operationId: 'updateAlertConfiguration',
        tag: 'Alert Configurations',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'release',
            changes: [
              {
                change: 'resource version added.',
                changeCode: 'resource-version-added',
                backwardCompatible: false,
              },
              {
                change:
                  "for the 'query' request parameter 'clusterName', default value was changed from 'default' to 'cluster0'",
                changeCode: 'request-parameter-default-value-changed',
                backwardCompatible: false,
              },
              {
                change: "added the optional property 'links' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'deprecate',
            sunsetDate: '2024-02-01',
            changes: [
              {
                change: 'deprecated by resource version 2023-02-01 and marked for removal for 2024-02-01.',
                changeCode: 'resource-version-deprecated',
                backwardCompatible: false,
              },
            ],
          },
        ],
      },
      {
        path: '/api/atlas/v2/groups/{groupId}/accessList',
        httpMethod: 'POST',
        operationId: 'createProjectIpAccessList',
        tag: 'Project IP Access List',
        versions: [
          {
            version: '2023-01-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change: "added the optional property 'name' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
              {
                change: "added the optional property 'comment' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
  {
    date: '2023-02-22',
    paths: [
      {
        path: '/api/atlas/v2/groups/byName/{groupName}',
        httpMethod: 'GET',
        operationId: 'getProjectByName',
        tag: 'Projects',
        versions: [
          {
            version: '2023-02-01',
            stabilityLevel: 'stable',
            changeType: 'update',
            changes: [
              {
                change:
                  "added the optional property 'withDefaultAlertsSettings' in the response with the '200' status.",
                changeCode: 'response-optional-property-added',
                backwardCompatible: true,
              },
            ],
          },
        ],
      },
    ],
  },
];

export const mockDiff = [
  {
    path: '/api/atlas/v2/federationSettings/{federationSettingsId}/connectedOrgConfigs',
    httpMethod: 'GET',
    operationId: 'listConnectedOrgConfigs',
    tag: 'Federated Authentication',
    changeType: 'update',
    changes: [
      {
        change: "removed the non-success response with the status '400'.",
        changeCode: 'response-non-success-status-removed',
        backwardCompatible: false,
      },
      {
        change: "added the new optional 'query' request parameter 'envelope'.",
        changeCode: 'new-optional-request-parameter',
        backwardCompatible: true,
      },
    ],
  },
  {
    path: '/api/atlas/v2/groups/{groupId}/accessList',
    httpMethod: 'POST',
    operationId: 'createProjectIpAccessList',
    tag: 'Project IP Access List',
    changeType: 'removed',
    changes: [
      {
        change: "removed the required property 'name' from the response with the '200' status.",
        changeCode: 'response-required-property-removed',
        backwardCompatible: false,
      },
    ],
  },
];

export const mockIndex = {
  runDate: '2023-04-03',
  runId: '20230403105948111',
  specRevision: '422ab47f864909549362a7b39688404c82e2540b',
  specRevisionShort: '422ab47f864',
  versions: ['2023-01-01', '2023-02-01', '2023-08-11~preview'],
};
