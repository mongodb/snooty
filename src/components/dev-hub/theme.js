/**
 * @type {Object.<string, string>}
 * @property {string} fontsize returns px value
 */
const fontSize = {
  xsmall: '10px',
  tiny: '12px',
  small: '14px',
  default: '16px',
  h1: '48px',
  h2: '30px',
  h3: '24px',
  h4: '20px',
  tabNav: '10px',
};
/**
 * @type {Object}
 * @property {string} size returns px value
 * @property {function(string): number} stripUnit removes px unit
 */
const size = {
  tiny: '5px',
  small: '10px',
  default: '16px',
  medium: '24px',
  large: '32px',
  xlarge: '64px',
  xxlarge: '128px',
  maxWidth: '1400px',
};
const colorMap = {
  black: '#1B272D',
  darkGreen: '#13AA52',
  devBlack: '#1a282e',
  devWhite: '#e3ecea',
  greyDarkOne: '#5B6D75',
  greyDarkTwo: '#3A5058',
  greyDarkThree: '#2A3D46',
  greyLightOne: '#E3ECEA',
  greyLightTwo: '#B4C4C1',
  greyLightThree: '#85979A',
  lightGreen: '#0AD05B',
  orange: '#ED7E22',
  magenta: '#CD509B',
  salmon: '#E55F55',
  teal: '#2F9FC5',
  violet: '#6E60F9',
  white: '#FFFFFF',
  yellow: '#FDDC49',
};

/**
 * store common responsive sizes
 * @type {Object.<string, string>}
 */
const screenSize = {
  upToXSmall: 'only screen and (max-width: 374px)',
  xSmallAndUp: 'not all and (max-width: 374px)',
  upToSmall: 'only screen and (max-width: 420px)',
  smallAndUp: 'not all and (max-width: 420px)',
  upToMedium: 'only screen and (max-width: 767px)',
  mediumAndUp: 'not all and (max-width: 767px)',
  upToLarge: 'only screen and (max-width: 1023px)',
  largeAndUp: 'not all and (max-width: 1023px)',
};

export { colorMap, fontSize, screenSize, size };
