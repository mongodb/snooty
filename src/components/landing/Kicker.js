import React from 'react';
import PropTypes from 'prop-types';
import ComponentFactory from '../ComponentFactory';
import { css } from '@emotion/core';
import { uiColors } from '@leafygreen-ui/palette';
import { Overline } from '@leafygreen-ui/typography';

const Kicker = ({ nodeData: { argument }, ...rest }) => (
  <Overline
    css={css`
      color: ${uiColors.gray.dark1};
    `}
  >
    {argument.map((child, i) => (
      <ComponentFactory {...rest} nodeData={child} key={i} />
    ))}
  </Overline>
);

Kicker.propTypes = {
  nodeData: PropTypes.shape({
    argument: PropTypes.arrayOf(PropTypes.object).isRequired,
  }).isRequired,
};

export default Kicker;
