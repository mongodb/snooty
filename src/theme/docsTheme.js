/**
 * @type {Object.<string, string>}
 * @property {string} fontsize returns px value
 */
const fontSize = {
  xsmall: '11px',
  tiny: '12px',
  small: '14px',
  default: '16px',
  h1: '36px',
  h2: '24px',
  h3: '18px',
  h4: '16px',
};
/**
 * @type {Object}
 * @property {string} size returns px value
 * @property {function(string): number} stripUnit removes px unit
 */
const size = {
  tiny: '4px',
  small: '8px',
  default: '16px',
  medium: '24px',
  large: '32px',
  xlarge: '64px',
  xxlarge: '128px',
  maxWidth: '1400px',
  /** @type {function(string): number} */
  stripUnit(unit) {
    return parseInt(unit, 10);
  },
};
/**
 * store common responsive sizes
 * @type {Object.<string, string>}
 */
const screenSize = {
  upToXSmall: 'only screen and (max-width: 320px)',
  xSmallAndUp: 'not all and (max-width: 320px)',
  upToSmall: 'only screen and (max-width: 420px)',
  smallAndUp: 'not all and (max-width: 420px)',
  upToMedium: 'only screen and (max-width: 767px)',
  mediumAndUp: 'not all and (max-width: 767px)',
  upToLarge: 'only screen and (max-width: 1023px)',
  largeAndUp: 'not all and (max-width: 1023px)',
  upToXLarge: 'only screen and (max-width: 1200px)',
  xLargeAndUp: 'not all and (max-width: 1200px)',
  tablet: 'only screen and (min-width: 421px) and (max-width: 1023px)',
};

const header = {
  bannerHeight: '40px',
  navbarHeight: '45px',
};

const transitionSpeed = {
  contentFadeOut: '100ms',
  contentFadeIn: '200ms',
};

export const theme = {
  fontSize,
  header,
  screenSize,
  size,
  transitionSpeed,
};
