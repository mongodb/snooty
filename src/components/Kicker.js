import React from 'react';
import PropTypes from 'prop-types';
import styled from '@emotion/styled';
import { Overline } from '@leafygreen-ui/typography';
import { uiColors } from '@leafygreen-ui/palette';
import { theme } from '../theme/docsTheme';
import ComponentFactory from './ComponentFactory';

const StyledKicker = styled(Overline)`
  grid-column: 2;
  font-size: ${theme.fontSize.small};
  color: ${uiColors.gray.dark1};
  padding-top: 80px;
  padding-bottom: ${theme.size.small};
  @media ${theme.screenSize.upToSmall} {
    padding-top: 56px;
  }
  @media ${theme.screenSize.upToXSmall} {
    padding-top: ${theme.size.large};
  }
`;

const Kicker = ({ nodeData: { argument }, ...rest }) => {
  return (
    <StyledKicker>
      {argument.map((child, i) => (
        <ComponentFactory {...rest} nodeData={child} key={i} />
      ))}
    </StyledKicker>
  );
};

Kicker.propTypes = {
  nodeData: PropTypes.shape({
    argument: PropTypes.arrayOf(PropTypes.object).isRequired,
  }).isRequired,
};

export default Kicker;
