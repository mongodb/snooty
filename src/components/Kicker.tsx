import React from 'react';
import { css } from '@leafygreen-ui/emotion';
import { theme } from '../theme/docsTheme';
import { Directive } from '../types/ast';
import ComponentFactory from './ComponentFactory';
import Overline from './Internal/Overline';

const kickerBaseStyle = css`
  grid-column: 2;
  @media ${theme.screenSize.upToSmall} {
    padding-top: 56px;
  }
  @media ${theme.screenSize.upToXSmall} {
    padding-top: ${theme.size.large};
  }
`;

const Kicker = ({ nodeData: { argument }, ...rest }: { nodeData: Directive }) => {
  return (
    <Overline className={kickerBaseStyle}>
      {argument.map((child, i) => (
        <ComponentFactory {...rest} nodeData={child} key={i} />
      ))}
    </Overline>
  );
};

export default Kicker;
