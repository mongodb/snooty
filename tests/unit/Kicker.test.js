import React from 'react';
import { mount } from 'enzyme';
import { ThemeProvider } from 'emotion-theming';
import Kicker from '../../src/components/landing/Kicker';
import { theme } from '../../src/theme/docsTheme';

// data for this component
import mockData from './data/landing/Kicker.test.json';

it('renders correctly', () => {
  const tree = mount(
    <ThemeProvider theme={theme}>
      <Kicker nodeData={mockData} />
    </ThemeProvider>
  )
    .childAt(0)
    .childAt(0);
  expect(tree).toMatchSnapshot();
});
