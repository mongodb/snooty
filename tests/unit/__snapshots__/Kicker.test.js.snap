// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
.emotion-2 {
  margin: unset;
  font-family: Akzidenz,'Helvetica Neue',Helvetica,Arial,sans-serif;
  color: #21313C;
  font-size: 12px;
  font-weight: 600;
  text-transform: uppercase;
  line-height: 16px;
  -webkit-letter-spacing: 0.4px;
  -moz-letter-spacing: 0.4px;
  -ms-letter-spacing: 0.4px;
  letter-spacing: 0.4px;
}

.emotion-0 {
  font-size: 14px;
  color: #5D6C74;
  padding-top: 64px;
  padding-bottom: 8px;
}

@media only screen and (max-width:767px) {
  .emotion-0 {
    padding-top: 32px;
  }
}

@media only screen and (max-width:420px) {
  .emotion-0 {
    padding-top: 24px;
  }
}

<StyledKicker>
  <Overline
    className="emotion-0 emotion-1"
  >
    <Box
      className="emotion-0 emotion-1 emotion-2"
    >
      <div
        className="emotion-0 emotion-1 emotion-2"
      >
        <ComponentFactory
          key="0"
          nodeData={
            Object {
              "position": Object {
                "start": Object {
                  "line": 111,
                },
              },
              "type": "text",
              "value": "Kicker text",
            }
          }
        >
          <Text
            nodeData={
              Object {
                "position": Object {
                  "start": Object {
                    "line": 111,
                  },
                },
                "type": "text",
                "value": "Kicker text",
              }
            }
          >
            Kicker text
          </Text>
        </ComponentFactory>
      </div>
    </Box>
  </Overline>
</StyledKicker>
`;
